keps/sig-network/20200103-mixed-protocol-lb.md

```diff
@@ -0,0 +1,351 @@
+
+Summary: The implementation of this feature can affect the bills of GCP users. However the following perspectives are also observed:
+- if a user wants to have UDP and TCP ports behind the same NLB two Services with must be defined, one for TCP and one for UDP. As the pricing is based on the number of forwarding rules this setup also means the same pricing as with the single Service instance.
+- if a user is happy with 2 NLB (Service) instances for TCP and UDP still the user has two more forwarding rules to be billed - i.e. it has the same effect on pricing as if those TCP and UDP endpoints were behind the same NLB instance
+- already now the bills of users is affected if they have more than 5 forwarding rules as the result of their current Service definitions (e.g. 6 or more port definitions in a single Serice, or if the number of all port definitions in different Services is 6 or more, etc.)
+
+That is, if we consider the "single Service with 6 ports" case the user has to pay more for that single Service ("single LB instance") than for another Service (another LB instance) with 5 or less ports already now. It is not the number of LBs that matters. This phenomenon is already there with the current practice, and the enabling of mixed protocols will not change it to the worse.
+
+#### IBM Cloud
+
+The IBM Cloud CPI implementation supports TCP, UDP, HTTP protocol values in K8s Services, and it supports mutiple protocols in a Service. The IBM Cloud CPI creates VPC Load Balancer in a VPC based cluster and NLB in a classic cloud based cluster.
+
+The VPC Load Balancer supports TCP and HTTP, and it is possible to create TCP and HTTP listeners for the same LB instance. UDP is not supported. 
+
+The VPC LB pricing is time and data amount based, i.e. the number of protocols on the same LB instance does not affect it.
+
+The NLB supports TCP and UDP on the same NLB instance. The usage of NLB does not have pricing effects, it is part of the IKS basic package.
+
+Summary: once this feature is implemented IBM Cloud VPC LB can use TCP and HTTP ports from a single Service definition. NLB can use TCP and UDP ports from a single Service definition.
+
+#### OpenStack
+
+The OpenStack CPI supports TCP, UDP, HTTP(S) in Service definitions and can configure the Octavia listeners with the protocols defined in the Service.
+OpenStack Octavia supports TCP, UDP and HTTP(S) on listeners, an own listener must be configured for each protocol, and different listeners can be used on the same LB instance.
+
+Summary: the OpenStack based clouds that use Octavia v2 as their LBaaS seems to support this feature once implemented. Pricing is up to their model.
```

Comment:
[Octavia version](https://github.com/openstack/octavia/commit/b42a64a00ef549c69bc6835db9234a889b5ba62c) should >= 5.0.0 which first release in [OpenStack Train](https://releases.openstack.org/train/index.html#train-octavia)

Reply:
@thinalai Thank you for the comment! Indeed, I can see this bugfix in the [5.0.0 release notes ](https://docs.openstack.org/releasenotes/octavia/train.html#relnotes-5-0-0-stable-train)
`Fixed bug which prevented the creation of listeners for different protocols on the same port (i.e: tcp port 53, and udp port 53).`
I add this note to the KEP.